### DISCUSSION TITLE
Bufferline Broken?

### DISCUSSION THREAD

**Discussion by directormac** [LazyVim/LazyVim#3239] [General]
```txt
   Error  09:07:17 PM msg_show.lua_error E5108: Error executing lua ...nvim/lazy/bufferline.nvim/lua/bufferline/diagnostics.lua:75: attempt to call field 'is_enabled' (a nil value)
stack traceback:
	...nvim/lazy/bufferline.nvim/lua/bufferline/diagnostics.lua:75: in function <...nvim/lazy/bufferline.nvim/lua/bufferline/diagnostics.lua:71>
	...nvim/lazy/bufferline.nvim/lua/bufferline/diagnostics.lua:139: in function 'get'
	...are/nvim/lazy/bufferline.nvim/lua/bufferline/buffers.lua:63: in function 'get_components'
	...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:56: in function <...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:54>
```

is it just me or someone else is experiencing it too?

**Comment by dpetka2001** [comment]
Check if you're on old nightly Neovim version and upgrade if necessary.

**Reply by directormac** [reply]
Nah i didn't update on v0.10 on neovim yet, 0.9.5 breaking current lazyvim config i think. last night it was the indent line for me. i guess i'll update to 0.10 now.

**Reply by folke** [reply]
It seems that bufferline made a change for 0.10 that is not backward compatible with 0.9.5. 
Best to open an issue there if it doesnt exist yet

**Reply by directormac** [reply]
Hey @folke ! i also tried updating on 0.10 it seems that it only exists on 0.9.5, When i upgrade on 0.10 i have tsserver errors, im in the middle of the project now everytime i type nvim keeps notifying about cmp errors. sigh. i will post details later.

**Reply by folke** [reply]
Buffer line just pushed a fix I think

**Reply by directormac** [reply]
Actually [this](https://github.com/akinsho/bufferline.nvim/commit/2cd39842c6426fb6c9a79fa57420121cc81c9804) broke it for 0.9.5

**Comment by directormac** [comment]
fix for 0.9.5 

```lua
 {
    "akinsho/bufferline.nvim",
    commit = "73540cb95f8d95aa1af3ed57713c6720c78af915",
}
```

**Comment by folke** [comment]
I just pushed an update to LazyVim that pins bufferline to the last working commit on 0.9.x.
You'll have to run `:Lazy` and  update to downgrade to the older version

**Reply by directormac** [reply]
as always man you are amazing! i will also try 0.10 also soon seems you made a fix on cmp thing, whatabout the tsserver on 0.10.2?

**Reply by directormac** [reply]
here is the error on 0.10 specifically on svelte file, i think their lsp needs an update.

```txt
   Error  02:04:28 AM msg_show.lua_error Error executing vim.schedule lua callback: ...local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/snippet.lua:409: snippet parsing failed.
stack traceback:
	[C]: in function 'error'
	...local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/snippet.lua:409: in function 'parse'
	...rtifex/.local/share/nvim/lazy/nvim-cmp/lua/cmp/entry.lua:130: in function 'callback'
	.../.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/cache.lua:38: in function 'get_word'
	...rtifex/.local/share/nvim/lazy/nvim-cmp/lua/cmp/entry.lua:81: in function 'callback'
	.../.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/cache.lua:38: in function 'get_offset'
	...tifex/.local/share/nvim/lazy/nvim-cmp/lua/cmp/source.lua:353: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
   Error  02:04:28 AM notify.error ...local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/snippet.lua:409: snippet parsing failed.
   Error  02:04:29 AM notify.error ...local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/snippet.lua:409: snippet parsing failed.
 ```

**Reply by folke** [reply]
I've just pushed some changes that may fix that snippet issue using native snippets. 
It's currently only available on main. (see https://www.lazyvim.org/configuration/lazy.nvim)

The problem seems to be that native snippets don't support recursive placeholders.

My fix should counter that and show a warning when the fix was needed.

Would be great if you could test and see if your problem is also resolved this way.

**Reply by directormac** [reply]
Hi folke i still have the same errors when editing svelte although the tsserver error is now good. still that the same error.

**Reply by folke** [reply]
Can you show me the exact snippet in the error notification?
![image](https://github.com/LazyVim/LazyVim/assets/292349/4ec59c5e-6dda-4a44-905c-1a8031b1eff2)

Like in that one, but if it didnt work for svelte, you'd probably get an error instead of a warning.

Please copy/paste the snippet in that notification

**Reply by folke** [reply]
I created a tracking issue on Neovim, please add your snippet there https://github.com/neovim/neovim/issues/28858

**Reply by directormac** [reply]
[This](https://github.com/LazyVim/LazyVim/discussions/3239#discussioncomment-9489401) is the exact error i am getting on svelte files havent checked other, earlier it was about tsserver now its just svelte files failing
